base/applications/mspaint/winproc.cpp

```diff
@@ -575,6 +576,20 @@ LRESULT CMainWindow::OnSysColorChange(UINT nMsg, WPARAM wParam, LPARAM lParam, B
     return 0;
 }
 
+static CStringW EscapeURLW(LPCWSTR psz)
+{
+    char utf8[256];
+    ::WideCharToMultiByte(CP_UTF8, 0, psz, -1, utf8, _countof(utf8), NULL, NULL);
+
+    char url[256];
+    DWORD cch = _countof(url);
+    UrlEscapeA(utf8, url, &cch, URL_ESCAPE_PERCENT | URL_ESCAPE_SEGMENT_ONLY);
+
+    WCHAR wide[256];
+    ::MultiByteToWideChar(CP_ACP, 0, url, -1, wide, _countof(wide));
+    return CStringW(wide);
```

Comment:
I don't understand why you go to UTF8 then go back to UTF16, instead of just calling `UrlEscapeW` .

Reply:
Any care to explain instead of closing as resolved?